test/ruby/test_hash.rb

```diff
@@ -282,6 +282,15 @@ def test_ASET_fstring_key
     assert_same a.keys[0], b.keys[0]
   end
 
+  def test_ASET_fstring_key_frozen
+    a, b = {}, {}
+    k1 = "abc"
+    k2 = "abc"
+    assert_equal 1, a[k1.freeze] = 1
+    assert_equal 1, b[k2.freeze] = 1
```

Comment:
As an assignment expression always returns the RHS, these assertions are unrelated with `Hash#[]=`.

Reply:
I copy pasted that from the existing test above, but yes, it's a bit silly I'll clean it up.